addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,97 @@
+    allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('qr', 'QR')], default='all')
+
+    def _get_payment_terminal_selection(self):
+        selection = super()._get_payment_terminal_selection()
+        selection += [('paytm', 'PayTM')]
+        return selection
+
+    def _paytm_make_request(self, url, payload=None):
+        """ Make a request to PayTM API.
+
+        Note: self.ensure_one()
+
+        :param str url: The url to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        self.ensure_one()
+        try:
+            response = requests.post(url, json=payload)
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            _logger.warning("Cannot connect with PayTM")
+            return {'body':{'resultInfo': {'resultCode': "F", "resultMsg": _("Unable to establish connection with PayTM.")}}}
+        return response.json()
+
+    def paytm_make_payment_request(self, amount, transaction, timestamp):
+        body = self._paytm_get_request_body(transaction, timestamp)
+        body['transactionAmount'] = str(int(amount))
+        if self.accept_payment == 'auto':
+            body['autoAccept'] = 'True'
+        body['paymentMode'] = self.allowed_payment_modes.upper()
+        head = self._paytm_get_request_head(body)
+        merchantExtendedInfo = {'paymentMode': self.allowed_payment_modes.upper()}
+        if self.accept_payment == 'auto':
+            merchantExtendedInfo['autoAccept'] = 'True'
+        body['merchantExtendedInfo'] = merchantExtendedInfo
+        payload = {'head': head, 'body': body}
+        api_url = 'https://securegw-stage.paytm.in/ecr/payment/request' if self.paytm_test_mode else 'https://securegw-edc.paytm.in/ecr/payment/request'
+        response = self._paytm_make_request(api_url, payload=payload)
+        return response
+
+    def paytm_fetch_payment_status(self, transaction, timestamp):
+        body = self._paytm_get_request_body(transaction, timestamp)
+        head = self._paytm_get_request_head(body)
+        payload = {'head': head, 'body': body}
+        api_url = 'https://securegw-stage.paytm.in/ecr/V2/payment/status' if self.paytm_test_mode else 'https://securegw-edc.paytm.in/ecr/V2/payment/status'
+        response = self._paytm_make_request(api_url, payload=payload)
+        return response
+
+    def _paytm_get_request_body(self, transaction, timestamp):
+        time = datetime.fromtimestamp(timestamp)
+        utc = time.replace(tzinfo=tz.gettz('UTC'))
+        ist = utc.astimezone(tz.gettz('Asia/Kolkata'))
+        time = ist.strftime("%Y-%m-%d %H:%M:%S")
+        return {
+            'paytmMid': self.paytm_mid,
+            'paytmTid': self.paytm_tid,
+            'transactionDateTime': time,
```

Comment:
you do a lot of manipulation with the date and times that I don't understand
why not just take the date and time from the browser?
and reformat it if necessary ?
I imagine legal obligations
Can you explain to me please ?

Reply:
Ok, we will reformat the date and time from browser as necessary.
The manipulation was mainly done as requests to the paytm api required the date and time according to IST (Indian Standard Time) and to be consistent with their server time.

Reply:
Ok so take the IST time and format it
2 lines no ?